src/arc_dst.rs

```diff
@@ -0,0 +1,476 @@
+#![deny(missing_docs)]
+
+use std::{
+    borrow::{Borrow, Cow},
+    hash::Hash,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use dashmap::mapref::entry::Entry;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Deserializer};
+
+use super::arc::{ArcIntern, BoxRefCount, RefCount};
+
+impl<T: Copy> RefCount<[T]> {
+    fn from_slice(slice: &[T]) -> Box<RefCount<[T]>> {
+        use std::alloc::Layout;
+        let layout = Layout::new::<RefCount<()>>()
+            .extend(Layout::array::<T>(slice.len()).unwrap())
+            .unwrap()
+            .0
+            .pad_to_align();
+        // SAFETY: the layout is not zero-sized, it at least has an `AtomicUsize`.
+        let ptr = unsafe { std::alloc::alloc(layout) };
+        // imitate the `std::Arc::new_zeroed_slice` and `std::Arc::try_allocate_for_layout`.
+        let ptr =
+            std::ptr::slice_from_raw_parts_mut(ptr as *mut T, slice.len()) as *mut RefCount<[T]>;
+
+        unsafe {
+            // SAFETY: the ptr is allocated by the global allocator with proper layout, as per the
+            // [Memory Layout](https://doc.rust-lang.org/stable/std/boxed/index.html#memory-layout) section
+            // of [`std::boxed`].
+            let mut this = Box::from_raw(ptr);
+
+            std::ptr::write(&mut this.count, AtomicUsize::new(1));
+
```

Comment:
This could be done safely with
```suggestion

            this.as_ref().count.store(1, Ordering::Relaxed);

```
which could allow you to have just two lines of `unsafe`, right?

Reply:
I originally wrote as suggested, but when plagiarizing std code, I noticed std wrote this way.

According to `std::ptr::write`, *This is appropriate for initializing uninitialized memory*, and it provides more guarantees for droppings. Personally I suspect on some platform atomic usize might not be so primitive and calling .store on uninitilized memory might cause problems.